packages/date/src/DateRange.js

```diff
@@ -159,7 +158,7 @@ const DateRange = ({
 
   const onFocusChange = async input => {
     if (!input) await setFieldTouched(name, true);
```

Comment:
So the following should work if you can fix it. We need to only run `setFieldTouched` here if we don't have an input and autoSync is not enabled so `!input && !autoSync`. This will make sure we don't call validation too early before `syncDates` has finished.